Lib/test/test_wave.py

```diff
@@ -196,6 +197,22 @@ def test_read_wrong_sample_width(self):
         with self.assertRaisesRegex(wave.Error, 'bad sample width'):
             wave.open(io.BytesIO(b))
 
+    @skip_unless_working_chmod
+    def test_write_to_protected_file(self):
+        # gh-136523: Wave_write.__del__ should not throw
+        stderr = io.StringIO()
+        sys.stderr = stderr
+        try:
+            try:
+                with unwritable_filepath() as path:
+                    with wave.open(path, "wb"):
+                        pass
+            except PermissionError:
```

Comment:
PermissionError does not occur if the test is run as root.

But it should always fail if the path is an existing directory or if the parent path is not existing.